### ISSUE TITLE
Why hasn't v1.0.0 been published to crates.io?

### ISSUE THREAD

**Issue by Folyd** [atroche/rust-headless-chrome#323]
https://github.com/atroche/rust-headless-chrome/releases/tag/v1.0.0 has been released a couple of months, but it is still v0.9 on crates.io.

**Comment by llGaetanll** [comment]
For anyone else facing this problem, you can still use v1 with cargo by setting your dependency to the following:
```toml
[dependencies]
headless_chrome = { git = "https://github.com/atroche/rust-headless-chrome", rev = "fd9ba91" }
```
Where `fd9ba91` is the commit hash for v1

**Comment by Nemo157** [comment]
Or better is to put the tag in your `Cargo.toml`:

```toml
headless_chrome = { git = "https://github.com/atroche/rust-headless-chrome", tag = "v1.0.0" }
```

(which will still put the full revision into your `Cargo.lock` to protect against the mutability of tags).

**Comment by elpiel** [comment]
I would love to see v1.0 released on crates.io too!

**Comment by mdrokz** [comment]
> I would love to see v1.0 released on crates.io too!
> 
> PS: It's interesting that **Cargo.toml version is still at 0.9** on the v1.0.0 tag.

Unfortunately I don't have access to the crates.io and so i never bothered to update it

**Comment by elpiel** [comment]
@atroche would you give us an update on this?
Do you plan to continue support and releasing of the crate or maybe you can allow someone else to push new releases like @mdrokz?

**Comment by Billy-Sheppard** [comment]
@mdrokz 

At what point do you (as you seem to be the most active reviewer/merger) consider forking this and republishing to crates.io?

New names to consider could be:
- `rusty-chrome`
- `chrome-oxidiser`
- `acephalous-chrome`
- `acephalous`

**Comment by mdrokz** [comment]
> @mdrokz 
> 
> At what point do you (as you seem to be the most active reviewer/merger) consider forking this and republishing to crates.io?
> 
> New names to consider could be (any and all could be pre- or suffixed with `rs`:
> - `rusty-chrome`
> - `chrome-oxidiser`
> - `acephalous-chrome`
> - `acephalous`

Im not experienced with how it would be done, could you explain the process ? Because forks can't have issues so will the issues be created on main repo and then the PR will come on the fork ?

**Comment by Nemo157** [comment]
You can enable issues on a fork from the repository settings. There are scripts you could use to clone the open issues over too (I don't know any specifically, I've just seen them used on other forks).

**Comment by elpiel** [comment]
Another option would be to push the code to a new repo instead of making a fork.
On forks it would always default to making prs to the main repo (not your fork)

**Comment by Sytten** [comment]
Did anyone try to contact the owner @atroche by email maybe first?
It is preferable to keep the name since it has recognition.

**Comment by mdrokz** [comment]
> Another option would be to push the code to a new repo instead of making a fork.
> On forks it would always default to making prs to the main repo (not your fork)

Im thinking what would be easier for existing users of the library ?

**Comment by Billy-Sheppard** [comment]
I agree it would be preferable to keep this repo.

I also agree it would be preferable for it to be a logical fork (new repo and code) rather than a literal one (pressing the fork button).

I am happy to be the one who moves it over and I can set up access to its crates.io and github repo for mdrokz at the minimum.

I will also try getting a hold of atroche on email however I suspect during the holiday period that may be more difficult than normal. 

In terms of redirecting users, mdrokz can add a line at the top of the README.md but there isn't a lot we could do beyond that.

**Comment by Billy-Sheppard** [comment]
FYI for everyone here, atroche has given me access to the crates.io and I have passed that on to mdrokz too. It may be worth considering moving the repo at some point in the future but for now it makes sense to leave it in place here.

@mdrokz I've invited you to the crates.io, can you please run a `cargo publish`? You're more familiar with the code, and I assume you consider it in a state worth releasing? Is it going to be released as `0.10.0` or `1.0.0`? I'll leave that up to you but I see no reason to not do the latter as indicated by this issue's title.

**Comment by mdrokz** [comment]
> FYI for everyone here, atroche has given me access to the crates.io and I have passed that on to mdrokz too. It may be worth considering moving the repo at some point in the future but for now it makes sense to leave it in place here.
> 
> @mdrokz I've invited you to the crates.io, can you please run a `cargo publish`? You're more familiar with the code, and I assume you consider it in a state worth releasing? Is it going to be released as `0.10.0` or `1.0.0`? I'll leave that up to you but I see no reason to not do the latter as indicated by this issue's title, and the most recent release/tag.

Sure im going to release as `1.0.0` on crates.io thank you so much for doing this @Billy-Sheppard

**Comment by elpiel** [comment]
@mdrokz I did not consider it fully ready for 1.0 but thanks for releasing it.

The docs build fails though:
https://docs.rs/crate/headless_chrome/1.0.0/builds/708635

Thanks @Billy-Sheppard for taking care of this.

**Comment by mdrokz** [comment]
> @mdrokz I did not consider it fully ready for 1.0 but thanks for releasing it.
> 
> The docs build fails though: https://docs.rs/crate/headless_chrome/1.0.0/builds/708635
> 
> Also, Thank you @Billy-Sheppard for taking care of this.

Ah i see, i will look into it thanks

**Comment by elpiel** [comment]
Also please consider updating the GH release because the 1.0 was released back in May and by the looks of things the latest crates release is based on your bump commit yesterday.
The same goes for the `1.0.0` tag as its last commit is from Feb (and not yesterday's commit)

**Comment by vvilhonen** [comment]
@mdrokz It seems the docs.rs build failed when trying to generate code outside of OUT_DIR, here's a PR for the project and I have rest of the docs fixes waiting until this gets merged and released to crates.io with a new version (which is not updated in the PR below)

https://github.com/mdrokz/auto_generate_cdp/pull/3

**Comment by vvilhonen** [comment]
PR pending on the previous PR https://github.com/atroche/rust-headless-chrome/pull/351

Would be good idea to tag this as 1.1.0 after merge and do a crates release with the same version. Auto_generate_cdp also uses ureq now instead of reqwest, which should speed up builds.

**Comment by vvilhonen** [comment]
There's still this during the build `warning: the following packages contain code that will be rejected by a future version of Rust: traitobject v0.1.0`. I wonder should this be fixed before 1.1.0? Tungstenite would probably be similar and not pulling tokio etc.

**Comment by Sytten** [comment]
@vvilhonen cant use tungstenite, check the other issue on that. But tldr it doesn't allow splitting the receiver and sender so the current architecture doesn't work.

**Comment by vvilhonen** [comment]
Ah, damn. Might be then good idea to release the current fixes and deal with later. I could take a crack at it soon if nobody figures it out before that 👍

**Comment by Billy-Sheppard** [comment]
This should all be fixed with the release of v1.0.2. Thanks for everyones help!
